bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/WakeOnLanUtility.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.lgwebos.internal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.InterfaceAddress;
+import java.net.NetworkInterface;
+import java.util.Enumeration;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class with utility functions to support Wake On Lan (WOL)
+ *
+ * @author Arjan Mels - Initial contribution
+ * @author Sebastian Prehn - Modification to getMACAddress
+ *
+ */
+@NonNullByDefault
+public class WakeOnLanUtility {
+
+    private static final Logger logger = LoggerFactory.getLogger(WakeOnLanUtility.class);
+
+    /**
+     * Get MAC address for host
+     * uses "arp" tool
+     *
+     * @param hostName Host Name (or IP address) of host to retrieve MAC address for
+     * @return MAC address
+     */
+    public static @Nullable String getMACAddress(String hostName) {
+        try {
+            Process proc = Runtime.getRuntime().exec("arping -r -c 1 -C 1 " + hostName); // arp on linux
+            int returnCode = proc.waitFor();
+            String s;
+            StringBuilder builder = new StringBuilder();
+            BufferedReader input = new BufferedReader(
+                    new InputStreamReader(returnCode == 0 ? proc.getInputStream() : proc.getErrorStream()));
+            while ((s = input.readLine()) != null) {
+                builder.append(s);
+            }
```

Comment:
Just in case you don't know, `readLine()` doesn't include the `\n` character in the result.

Reply:
That is fine in this case. We only expect one line or none.